src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/AutoConfirmUser/AutomaticallyConfirmOrganizationUsersValidator.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Bit.Core.AdminConsole.Enums;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies.PolicyRequirements;
+using Bit.Core.AdminConsole.Services;
+using Bit.Core.AdminConsole.Utilities.v2.Validation;
+using Bit.Core.Auth.UserFeatures.TwoFactorAuth.Interfaces;
+using Bit.Core.Billing.Enums;
+using Bit.Core.Enums;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using static Bit.Core.AdminConsole.Utilities.v2.Validation.ValidationResultHelpers;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.AutoConfirmUser;
+
+public class AutomaticallyConfirmOrganizationUsersValidator(
+    IOrganizationUserRepository organizationUserRepository,
+    IFeatureService featureService,
+    ITwoFactorIsEnabledQuery twoFactorIsEnabledQuery,
+    IPolicyService policyService,
+    IPolicyRequirementQuery policyRequirementQuery
+    ) : IAutomaticallyConfirmOrganizationUsersValidator
+{
+    public async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>> ValidateAsync(
+        AutomaticallyConfirmOrganizationUserRequestData request)
+    {
+        var orgUser = request.OrganizationUser;
+
+        if (orgUser.Status != OrganizationUserStatusType.Accepted)
+        {
+            return Invalid(request, new UserIsNotAccepted());
+        }
+
+        if (orgUser.OrganizationId != request.Organization.Id)
+        {
+            return Invalid(request, new OrganizationUserIdIsInvalid());
+        }
+
+        return await OrganizationUserOwnershipValidationAsync(request)
+            .ThenAsync(OrganizationUserConformsToOrganizationPoliciesAsync);
```

Comment:
(non-blocking clarification) Why the mixed approach? I guess because `ThenAsync` only accepts Tasks, and the first few checks are synchronous?

Reply:
Made some more methods. Should be all chained now !